SU-HSDO/suhumsci
# Feat(STN-690): Bug fix for raised vertical card width

# READY FOR REVIEW

## Summary
`hb-raised-cards-uniform-height` impacts both vertical linked cards and vertical cards in unexpected ways.
On vertical cards that contained inline items such as links or a vertical linked card, the display:flex applied to the card container from the utility class made it so the width of the cards was not full width of the space within a row.

Jira Card: https://sparkbox.atlassian.net/browse/STN-690

## Need Review By (Date)
3/18/21

## Urgency
medium

## Steps to Test
1. npm run test
2. Create a flexible page with a row of postcards with anchor links and a row of vertical linked cards where you only add the title and link. You can create additional rows with different types of content but those first two additions are important because the links are inline elements that will not naturally expand 100% to the container within `display:flex`.
![image](https://user-images.githubusercontent.com/9355698/111715917-5078d700-882b-11eb-888d-323a77ea0108.png)
(I added the orange background to `.hb-cards` for troubleshooting.)

- As you can see in the screenshot the cards with full paragraph content or an image, there is no issue with them going full width. However, the cards that are linked cards or only contain a short anchor link, which is similar to the examples shared in the Jira card as well.

3. Verify that this fix applies `width:100%` only to the div following the postcard container classes only and it does not trickle down any further into the elements.
The cards should look like this with the application to the card wrapping div.
![image](https://user-images.githubusercontent.com/9355698/111716169-d9900e00-882b-11eb-817e-586556feb0d8.png)

4. Verify this fix on both Traditional Colorful and that it doesn't affect horizontal cards negatively or other card types.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
